### PR TITLE
IDP changes to new framework

### PR DESCRIPTION
**Pull request by sagidayan**

### CODE REVIEW

File: provider/identityprovider/identity_provider_resource.go [hunk 1]
```diff
@@ -23,108 +23,122 @@ import (
...
 
-	// Create the resource:
-	result = &IdentityProviderResource{
-		collection: collection,
+	if diag.HasError() {
+		return
 	}
 
-	return
+	//TODO: add validations
```
Comment (bardielle) [inline provider/identityprovider/identity_provider_resource.go:137]: Please add the validation

Comment (sagidayan) [reply provider/identityprovider/identity_provider_resource.go:137]: I am uncertain if we need in this level. We have validations for all the rest. The only things without validations are `cluster` `id` and `name`...  Ill remove this function

Comment (sagidayan) [reply provider/identityprovider/identity_provider_resource.go]: On second thought, maybe we want to be sure just one of the idp "providers/methods" are passed. (this validation was not present in the previous implementation.

Comment (JohnStrunk) [reply provider/identityprovider/identity_provider_resource.go:137]: I don't want to hold up this PR, as the main focus is to get the code ported over.
This resource has many validations in it, and I think a fair number of them can be converted to use some of the framework's built-in validators.
Examples:
- htpassword username and password checks can use `stringvalidator.RegexMatches()` and the min/max length validators.
- The mappingmethod validator can be a `stringvalidator.OneOf()`

I suspect we can also cut a fair bit of logic in this and other resources by adding ConflictsWith() and/or ExactlyOneOf()

Comment (bardielle) [reply provider/identityprovider/identity_provider_resource.go:137]: Added a ticket https://issues.redhat.com/browse/OCM-3735

File: provider/identityprovider/identity_provider_resource.go [hunk 2]
```diff
@@ -391,13 +390,17 @@ func (r *IdentityProviderResource) Read(ctx context.Context, request resource.Re
...
+			state.Github.Teams = types.ListUnknown(types.StringType)
 		}
 		orgs, ok := githubObject.GetOrganizations()
 		if ok {
 			state.Github.Organizations, err = common.StringArrayToList(orgs)
 			if err != nil {
 				response.Diagnostics.AddError("failed to convert string slice to tf list", "GitHub Organizations conversion failed")
 			}
+		} else {
+			state.Github.Organizations = types.ListUnknown(types.StringType)
```
[the remark this answers is not in the document]
Comment (bardielle) [reply provider/identityprovider/identity_provider_resource.go:403]: Added a ticket for that: https://issues.redhat.com/browse/OCM-3734

**Pull request merged**